### PR TITLE
task_js_oop ( frogger)

### PR DESCRIPTION
**Pull request by andreymashko7**
### frogger-game

[Demo](https://andreymashko7.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/andreymashko7/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by andreymashko7** [comment]
self-check is done

File: submissions/AndriyM/task_js_oop/app.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        this.x = ENEMY_CONF.START_POSITION_X;
+    }
+    if (
+        player.x + ENEMY_CONF.PADDING_AVATAR_X > this.x &&
+        player.x < ENEMY_CONF.PADDING_AVATAR_X + this.x &&
+        player.y < ENEMY_CONF.PADDING_AVATAR_Y + this.y &&
+        player.y + ENEMY_CONF.PADDING_AVATAR_Y > this.y
+    ) {
+        alert("Game over! Score: 0");
+        player.reset();
```
Comment (OleksiyRudenko) [inline submissions/AndriyM/task_js_oop/app.js:51]: See items 4.iii and 4.iv of the requirements.

File: submissions/AndriyM/task_js_oop/app.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        y: FIELD_CONF.HEIGHT
+    }
+};
+
+const ENEMY_CONF = {
+    PADDING_AVATAR_X: 75,
+    PADDING_AVATAR_Y: 50,
+    START_POSITION_X: FIELD_CONF.ZERO_POSITION - FIELD_CONF.CELL_WIDTH,
+    END_POSITION_X: FIELD_CONF.WIDTH + FIELD_CONF.CELL_WIDTH,
+    START_POSITION_Y: [60, 145, 228],
```
Comment (OleksiyRudenko) [inline submissions/AndriyM/task_js_oop/app.js:27]: You are tasked to add three more rows for enemies.
Optimize this so it refers to something human-comprehensible (rows) and pixel calculations are done by a computer.

File: submissions/AndriyM/task_js_oop/app.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        allowedKey === PLAYER_CONF.MOVE_UP &&
+        this.y > FIELD_CONF.ZERO_POSITION
+    ) {
+        this.y -= FIELD_CONF.CELL_HEIGHT;
+    }
+    if (this.y < FIELD_CONF.WATTER_BLOCK) {
+        this.score++;
+        setTimeout(() => {
+            alert(`You win! Score: ${this.score}`);
+            player.reset();
```
Comment (OleksiyRudenko) [inline submissions/AndriyM/task_js_oop/app.js:99]: 4.iii

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@andreymashko7 very good job!
A couple of improvements and we're done

File: submissions/AndriyM/task_js_oop/app.js (hunk 3 above)
Comment (andreymashko7) [reply submissions/AndriyM/task_js_oop/app.js]: Please, review again

**Comment by andreymashko7** [comment]
Please, review again

File: submissions/AndriyM/task_js_oop/app.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+Enemy.prototype.checkCollisions = function () {
+	if (
+		this.player.x + ENEMY_CONF.PADDING_AVATAR_X > this.x &&
+		this.player.x < ENEMY_CONF.PADDING_AVATAR_X + this.x &&
+		this.player.y < ENEMY_CONF.PADDING_AVATAR_Y + this.y &&
+		this.player.y + ENEMY_CONF.PADDING_AVATAR_Y > this.y
+	) {
+		alert("Game over! Score: 0");
+		this.player.reset();
```
Comment (OleksiyRudenko) [inline submissions/AndriyM/task_js_oop/app.js:58]: It was good to have a setTimeout. You could keep both setTimeout and meet the requirement.
It was an opportunity to learn through questions.
There is still one. Ask in Students' chat, you will learn one fundamental thing about asynchronous code and use case for async operations in similar situations..

**Review by OleksiyRudenko — APPROVED** [review]
@andreymashko7 well done!

**Pull request merged**